app/controllers/api/messaging.php

```diff
@@ -2858,22 +2922,6 @@
         }
 
         if (!\is_null($targets)) {
-            $foundTargets = $dbForProject->find('targets', [
-                Query::equal('$id', $targets),
-                Query::equal('providerType', [MESSAGE_TYPE_SMS]),
-                Query::limit(\count($targets)),
-            ]);
-
-            if (\count($foundTargets) !== \count($targets)) {
-                throw new Exception(Exception::MESSAGE_TARGET_NOT_SMS);
-            }
-
-            foreach ($foundTargets as $target) {
-                if ($target->isEmpty()) {
-                    throw new Exception(Exception::USER_TARGET_NOT_FOUND);
-                }
-            }
-
```

Comment:
Why remove this validation?

Reply:
It's a heavy operation, we filter out any invalid targets in the worker, and other errors for message delivery are async